programs/benchmarks/noc/latency_aligned/noc_latency_aligned.c

```diff
@@ -0,0 +1,80 @@
+/**
+ * This program explores the absolute minimum amount of time that it can take to send one word and
+ * write it into a register on another core, when under the most favorable circumstances,
+ * and when controlling relative timing and optimizing the assembly.
+ */
+#include <stdint.h>
+#include <flexpret_io.h>
+#include <flexpret_noc.h>
+#include <stdlib.h>
+
+#include "align.h"
+
+#define N 100
+
+static int main_of(uint32_t core);
+
+static int send_main(uint32_t receiver);
+static int receive_main(uint32_t sender);
+
+int main() {
+    unsigned long coreid = read_csr(CSR_COREID);
+    srand(coreid);
+    if (coreid == 0) for (int i = 0; i < 10; i++) send_main(1);
+    if (coreid == 1) for (int i = 0; i < 10; i++) receive_main(0);
+}
+
+static int send_main(uint32_t receiver) {
+    asm volatile (
+        "li t4, 0x40000000\n\t"
+        WAIT_FOR_NEXT_ZERO_MOD_1024(send)  // clobber "a" registers, as well as t0, t1, t6
+        // like noc_send, but without blocking
+        "li t5, 0x1\n\t"  // noc destination
+        "sw t5, 8(t4)\n\t"
+        "li t5, 0x08\n\t"
+        "sw t5, 4(t4)\n\t"
+        "nop\n\t"
+        "nop\n\t"
+        "li t5, 42\n\t"  // Set noc data to 42
+        "sw t5, 8(t4)\n\t" // FIXME: Data must be written first? Why? Is it Hardware Bug?
```

Comment:
I recommend to record this in an issue.

Reply:
This question has since been answered -- it is just the way the interface is designed, where the address must be the last data to arrive so that the destination can be determined in that cycle. This has performance implications, but we might not keep this interface anyway.

Reply:
Even better!
```suggestion
        "sw t5, 8(t4)\n\t" // NOTE: Data must be written first. This is by design.
```